File: Human_written_code/Code_with_comments/CPP/Q30.cpp
#include <iostream>
#include <map>
using namespace std;

int main() {
    int t;
    cin >> t;  // Read number of test cases

    while (t--) {
        map<char, int> m;
        string n;
        cin >> n;  // Read input string
        int ans = 0;

        // Count frequency of each character in the string
        for (int i = 0; i < n.length(); i++) {
            m[n[i]]++;
        }

        // Check cases based on the number of unique characters
        if (m.size() == 4)
            cout << 4 << endl;  // All characters are unique
        else if (m.size() == 1)
            cout << -1 << endl;  // All characters are the same
        else if (m.size() == 3)
            cout << 4 << endl;  // Three unique characters
        else {
            // When there are two unique characters, check their counts
            for (auto it : m) {
                if (it.second == 3) {  // One character appears three times
                    cout << 6 << endl;
                    ans = 1;
                    break;
                }
            }
            if (ans == 0)
                cout << 4 << endl;  // Otherwise, both characters appear twice
        }
    }

    return 0;
}

File: Human_written_code/Code_with_comments/Python/Q30.py
def main():
    t = int(input().strip())
    
    for _ in range(t):
        s = input().strip()
        char_count = {}
        
        # Count occurrences of each character
        for char in s:
            char_count[char] = char_count.get(char, 0) + 1

        # Determine output based on the conditions
        if len(char_count) >= 3:
            print(4)
        elif len(char_count) == 2 and any(count == 2 for count in char_count.values()):
            print(4)
        elif len(char_count) == 2:
            print(6)
        else:
            print(-1)

if __name__ == "__main__":
    main()
